### PR TITLE
[Feature] compare with recipes

### PR DESCRIPTION
**Pull request by qrtt1**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**

feature added

**What this PR does / why we need it**:

New comparison strategy with the recipe configuration

**Which issue(s) this PR fixes**:

sc-30452

**Special notes for your reviewer**:

TBD

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

**Comment by popcornylu** [comment]
For branch switching logic, can we use merge-base, rather than branch latest commit?
```
COMMIT=$(git merge-base $(git rev-parse HEAD) main)     
```

github pull request use the three dot version diff. see [github compare branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests#three-dot-and-two-dot-git-diff-comparisons)

**Review by popcornylu — CHANGES_REQUESTED** [review]
1. If the git working directory is dirty, don't print the trackback
2. If there is no `command` or `commands` in piperider settings, the command shoud fail.
3. Remove the `--dbt-state target/` from the generated default recipe
4. `piperider compare --recipe xyz` should not show the dbt project select (if there are multiple dbt project)    i use ` git clone git@github.com:dbt-labs/dbt-core.git git_repo   ` in the git-repo-analytics repo

File: piperider_cli/configuration.py [hunk 1]
```diff
@@ -128,6 +126,8 @@ def from_dbt_project(cls, dbt_project_path, dbt_profiles_dir=None):
         credential = dbtutil.load_credential_from_dbt_profile(dbt_profile, profile_name, target_name)
         type_name = credential.get('type')
         dbt = {
+            'profile': profile_name,
+            'target': target_name,
```
[the remark this answers is not in the document]
Comment (kentwelcome) [reply piperider_cli/configuration.py]: fixed

**Comment by kentwelcome** [comment]
All the code review issues are fixed!

**Pull request merged**